document/cloud-client/analyze_form.py

```diff
@@ -0,0 +1,125 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def analyze_form(project_id, input_uri):
+    """Parse a form"""
+    # [START analyze_form]
+    from google.cloud import documentai_v1beta2 as documentai
+    from google.cloud.documentai_v1beta2 import types
+
+    client = documentai.DocumentUnderstandingServiceClient()
+
+    gcs_source = types.GcsSource(uri=input_uri)
+    input_config = types.InputConfig(
+        gcs_source=gcs_source, mime_type='application/pdf')
+
+    # Improve form parsing results by providing key-value pair hints.
+    # For each key hint, key is text that is likely to appear in the
+    # document as a form field name (i.e. "DOB").
+    # Value types are optional, but can be one or more of:
+    # ADDRESS, LOCATION, ORGANIZATION, PERSON, PHONE_NUMBER, ID,
+    # NUMBER, EMAIL, PRICE, TERMS, DATE, NAME
+    key_value_pair_hints = [
+        types.KeyValuePairHint(key='Name', value_types=['NAME']),
+        types.KeyValuePairHint(key='Agreement Number', value_types=['NUMBER'])
+    ]
+
+    # Setting enabled=True enables form extraction
+    form_extraction_params = types.FormExtractionParams(
+        enabled=True, key_value_pair_hints=key_value_pair_hints)
+
+    # For now, location must be us-central1
+    parent = "projects/{}/locations/us-central1".format(project_id)
+    request = types.ProcessDocumentRequest(
+        parent=parent,
+        input_config=input_config,
+        form_extraction_params=form_extraction_params)
+
+    return client.process_document(request=request)
+    # [END analyze_form]
+
+
+def parse_form_response(document):
+    """Parse Document response from the DAI API to make
+    it easier to work with. Returns:
+
+    [
+        {'name':
+            {'text': 'Address: ',
+            'start_index': 746,
+            'end_index': 755,
+            'confidence': 1.0,
+            'bounding_poly':
+                [x: 0.12581700086593628
+                y: 0.31313130259513855
+                , x: 0.19771242141723633
+                y: 0.31313130259513855
+                , x: 0.19771242141723633
+                y: 0.32575756311416626
+                , x: 0.12581700086593628
+                y: 0.32575756311416626]
+        },
+        'value':
+            {'text': '100 Franklin Street, Mountain View, CA, 94035\n',
+            'start_index': 755,
+            'end_index': 801,
+            'confidence': 1.0,
+            'bounding_poly':
+                [x: 0.2222222238779068
+                y: 0.31439393758773804
+                , x: 0.5947712659835815
+                y: 0.31439393758773804
+                , x: 0.5947712659835815
+                y: 0.32702019810676575
+                , x: 0.2222222238779068
+                y: 0.32702019810676575
+                ]
+        },
+        'page': 0},
+    ...
+    ]
+     """
```

Comment:
Instead of dropping the complete output. We try to decide what relevant parts we want to show via print statements.